libs/wingsdk/src/target-sim/resource.ts

```diff
@@ -277,7 +277,7 @@ export interface IResourceFactoryClient {
    * Function that will be called to create the resource.
    * @inflight
    */
-  handle(context: IResourceContext): Promise<IResource>;
+  handle(context: IResourceContext): Promise<IResource | undefined>;
```

Comment:
Can we add something to the docs maybe? As a rule of thumb when we're allowing callback functions to return nothing, I feel like it's worth making it clear what the nil behavior means

Reply:
Done